### PR TITLE
Honoring Context missing strategy for all TraceContext.GetEntity() call sites

### PR DESCRIPTION
**Pull request by yogiraj07**
*Description of changes:*
`AWSXRayRecorder.Instance.TraceContext.GetEntity()` call sites should be surrounded by `EntityNotAvailableException` catch block in order to honor ContextMissingStrategy configured to the X-Ray Recorder. Currently there are some calls that don't follow the pattern and this PR fies the issue.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: sdk/test/UnitTests/HttpWebRequestTracingExtensionTests.cs [hunk 1]
```diff
@@ -217,5 +217,32 @@ public async Task TestExceptionGetAsyncResponseTraced()
...
+#else
+            AWSXRayRecorder.InitializeInstance(recorder: _recorder);
+# endif
+            Assert.IsFalse(AWSXRayRecorder.Instance.IsTracingDisabled());
+
+            AWSXRayRecorder.Instance.ContextMissingStrategy = Core.Strategies.ContextMissingStrategy.LOG_ERROR;
+
+            _recorder.EndSegment();
+
+            // The test should not break. No segment is available in the context, however, since the context missing strategy is log error,
```
Comment (chrisradek) [inline sdk/test/UnitTests/HttpWebRequestTracingExtensionTests.cs]: Is there a way to detect if the logger prints the correct message?

Comment (yogiraj07) [reply sdk/test/UnitTests/HttpWebRequestTracingExtensionTests.cs]: Thats interesting point. However, there is no way to get the value from the logger message at the moment. I think, in future, we need to have custom logger implementation which has access to logger instance at AWSXRayRecorder level, to get the value of the message.

Comment (chrisradek) [reply sdk/test/UnitTests/HttpWebRequestTracingExtensionTests.cs]: Any chance we can rerun now that we have appveyor? :)

Comment (yogiraj07) [reply sdk/test/UnitTests/HttpWebRequestTracingExtensionTests.cs]: Sure, done.

**Pull request merged**